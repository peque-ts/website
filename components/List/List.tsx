import React from 'react';

interface Props<TItem> {
  items: TItem[];
  children: (item: TItem, index: number) => JSX.Element;
  className?: string;
}

export function List<TItem>({ items, children, className }: Props<TItem>) {
  return (
    <ul className={className}>
      {items.map((item, index) => (
        <li key={index}>{children(item, index)}</li>
      ))}
    </ul>
  );
}
