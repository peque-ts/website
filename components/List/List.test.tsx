import { render } from '@testing-library/react';

import { List } from './List';

describe('List', () => {
  it('should render', () => {
    const { asFragment } = render(
      <List items={['one', 'two', 'three']}>
        {(item, index) => (
          <span>
            {item} ({index})
          </span>
        )}
      </List>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
