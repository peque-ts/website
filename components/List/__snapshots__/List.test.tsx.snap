// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <span>
        one (0)
      </span>
    </li>
    <li>
      <span>
        two (1)
      </span>
    </li>
    <li>
      <span>
        three (2)
      </span>
    </li>
  </ul>
</DocumentFragment>
`;
